programs/drift/src/instructions/admin.rs

```diff
@@ -5614,3 +6514,325 @@ pub struct UpdateIfRebalanceConfig<'info> {
     )]
     pub state: Box<Account<'info, State>>,
 }
+
+#[derive(Accounts)]
+#[instruction(
+    name: [u8; 32],
+)]
+pub struct InitializeLpPool<'info> {
+    #[account(mut)]
+    pub admin: Signer<'info>,
+    #[account(
+        init,
+        seeds = [b"lp_pool", name.as_ref()],
+        space = LPPool::SIZE,
+        bump,
+        payer = admin
+    )]
+    pub lp_pool: AccountLoader<'info, LPPool>,
+
+    pub mint: Account<'info, anchor_spl::token::Mint>,
+
+    #[account(
+        init,
+        seeds = [b"LP_POOL_TOKEN_VAULT".as_ref(), lp_pool.key().as_ref()],
+        bump,
+        payer = admin,
+        token::mint = mint,
+        token::authority = drift_signer
+    )]
+    pub lp_pool_token_vault: Box<InterfaceAccount<'info, TokenAccount>>,
+
+    #[account(
+        init,
+        seeds = [AMM_MAP_PDA_SEED.as_ref(), lp_pool.key().as_ref()],
+        bump,
+        space = AmmConstituentMapping::space(0 as usize),
+        payer = admin,
+    )]
+    pub amm_constituent_mapping: Box<Account<'info, AmmConstituentMapping>>,
+
+    #[account(
+        init,
+        seeds = [CONSTITUENT_TARGET_BASE_PDA_SEED.as_ref(), lp_pool.key().as_ref()],
+        bump,
+        space = ConstituentTargetBase::space(0 as usize),
+        payer = admin,
+    )]
+    pub constituent_target_base: Box<Account<'info, ConstituentTargetBase>>,
+
+    #[account(
+        init,
+        seeds = [CONSTITUENT_CORRELATIONS_PDA_SEED.as_ref(), lp_pool.key().as_ref()],
+        bump,
+        space = ConstituentCorrelations::space(0 as usize),
+        payer = admin,
+    )]
+    pub constituent_correlations: Box<Account<'info, ConstituentCorrelations>>,
+
+    #[account(
+        has_one = admin
+    )]
+    pub state: Box<Account<'info, State>>,
+    /// CHECK: program signer
+    pub drift_signer: AccountInfo<'info>,
+
+    pub token_program: Program<'info, Token>,
+
+    pub rent: Sysvar<'info, Rent>,
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(Accounts)]
+#[instruction(
+    spot_market_index: u16,
+)]
+pub struct InitializeConstituent<'info> {
+    #[account()]
+    pub state: Box<Account<'info, State>>,
+    #[account(
+        mut,
+        constraint = admin.key() == admin_hot_wallet::id() || admin.key() == state.admin
+    )]
+    pub admin: Signer<'info>,
+
+    #[account(mut)]
+    pub lp_pool: AccountLoader<'info, LPPool>,
+
+    #[account(
+        mut,
+        seeds = [CONSTITUENT_TARGET_BASE_PDA_SEED.as_ref(), lp_pool.key().as_ref()],
+        bump = constituent_target_base.bump,
```

Comment:
does doing this speed up the pda derivation?

Reply:
should we just store the constituent target base key on the lp pool so we can just check equality for keys?